src/backend/types.rs

```diff
@@ -12,22 +12,163 @@ pub struct HeadInfo {
     pub l1_epoch: Epoch,
 }
 
-impl TryFrom<sled::IVec> for HeadInfo {
+impl TryFrom<Block<Transaction>> for HeadInfo {
```

Comment:
I think now that the backend module basically only has this single type, it makes sense to remove this module. I think we can move it to `driver` (maybe in its own submodule to keep things neat).

Reply:
Okis 👍🏼